source/common/common/key_value_store_base.cc

```diff
@@ -58,20 +70,31 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
   return true;
 }
 
-void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_view value_view) {
+void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_view value_view,
+                                    absl::optional<std::chrono::milliseconds> ttl) {
   ENVOY_BUG(!under_iterate_, "addOrUpdate under the stack of iterate");
   std::string key(key_view);
   std::string value(value_view);
+  // Remove ttl if <= 0
+  if (ttl && ttl <= std::chrono::milliseconds(0)) {
+    ASSERT(false);
```

Comment:
remove?

Reply:
This is Ryan's suggestion (hopefully) to use `ASSERT(false);` to throw an error in debug mode if an invalid ttl is passed in.